content/en/docs/concepts/workloads/controllers/replicaset.md

```diff
@@ -88,66 +102,114 @@ Events:
+```
 
-For labels, make sure to not overlap with other controllers. For more information, see [pod selector](#pod-selector).
+If we create the Pods first:
+```shell
+kubectl create -f http://k8s.io/examples/pods/pod-rs.yaml
+```
 
-For [restart policy](/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy), the only allowed value for `.spec.template.spec.restartPolicy` is `Always`, which is the default.
+And then create the ReplicaSet however:
+```shell
+kubectl create -f http://k8s.io/examples/controllers/frontend.yaml
+```
 
-For local container restarts, ReplicaSet delegates to an agent on the node,
-for example the [Kubelet](/docs/admin/kubelet/) or Docker.
+We shall see that the ReplicaSet has acquired the Pods and has only created new ones according to its spec until the
+number of its new Pods and the original matches its desired count. As fetching the pods:
+```shell
+kubectl get pods
+```
 
-### Pod Selector
+Will reveal in its output:
+```shell
+NAME             READY   STATUS    RESTARTS   AGE
+frontend-pxj4r   1/1     Running   0          5s
+pod1             1/1     Running   0          13s
+pod2             1/1     Running   0          13s
+```
 
-The `.spec.selector` field is a [label selector](/docs/concepts/overview/working-with-objects/labels/). A ReplicaSet
-manages all the pods with labels that match the selector. It does not distinguish
-between pods that it created or deleted and pods that another person or process created or
-deleted. This allows the ReplicaSet to be replaced without affecting the running pods.
+In this manner, a ReplicaSet can own a non-homogenous set of Pods
 
-The `.spec.template.metadata.labels` must match the `.spec.selector`, or it will
-be rejected by the API.
+## Writing a ReplicaSet Manifest
 
+As with all other Kubernetes API objects, a ReplicaSet needs the `apiVersion`, `kind`, and `metadata` fields.
+For ReplicaSets, the kind is always just ReplicaSet.
 In Kubernetes 1.9 the API version `apps/v1` on the ReplicaSet kind is the current version and is enabled by default. The API version `apps/v1beta2` is deprecated.
+Please refer to the first lines of the `frontend.yaml` example for guidance.
 
-Also you should not normally create any pods whose labels match this selector, either directly, with 
-another ReplicaSet, or with another controller such as a Deployment. If you do so, the ReplicaSet thinks that it 
-created the other pods. Kubernetes does not stop you from doing this.
+A ReplicaSet also needs a [`.spec` section](https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status).
+
+### Pod Template
 
-If you do end up with multiple controllers that have overlapping selectors, you
-will have to manage the deletion yourself.
+The `.spec.template` is a [pod template](/docs/concepts/workloads/pods/pod-overview/#pod-templates) which also requires
+to have labels in place. In our `frontend.yaml` example we had one label: `tier: frontend`.
```

Comment:
"which also requires labels."
or
"which is also required to have labels in place."